### PR TITLE
`isDateAvailable` option for DatePickers

### PR DESCRIPTION
**Pull request by Aeonax**

### CODE REVIEW

**Comment by Aeonax** [comment]
This setting will disable day in `DatePicker` from selection - show customers a day as unavailable on "picking" level, not after picking with words "this day is unavailable ....."

**Comment by fmnoise** [comment]
passing a function which will determine date availability is more convenient and functional approach than passing data for each day or I'm missing something?

File: pages/index.js [hunk 1]
```diff
@@ -231,7 +232,41 @@ class DatePickerDemo extends React.Component {
...
+        <div>
+          <h2>Limited by `isDateAvailable`</h2>
+          <DateSelectButton date={selectedDate}
+                            locale='en'
+                            wrapperStyle={{ width: 200, margin: SPACING.M }}
+                            onClick={this.toggleShowDatePicker}/>
+          {showDatePicker ? (
+            <DatePickerWithData
+              onClose={this.toggleShowDatePicker}
+              isDateAvailable={date => { var day = (new Date(date)).getDay(); return day != 0 && day != 6 }}
```
Comment (bskog) [inline pages/index.js:260]: Always triple equal in js: `return day !== 0 && day !== 6;`

File: pages/index.js [hunk 2]
```diff
@@ -231,7 +232,41 @@ class DatePickerDemo extends React.Component {
...
+              onDateClick={date => {
+                console.log("Selected Date: ", date);
```
Comment (bskog) [inline pages/index.js:262]: Remove console log?

**Comment by bskog** [comment]
> passing a function which will determine date availability is more convenient and functional approach than passing data for each day or I'm missing something?

Ah, I didn't see the `DatePickerWithData.js`. Thought we had to create and pass all date data to `DatePicker.js` anyways, but since that not necessary this make totally sense.

**Pull request merged**